scripts/toolchain/dbtenv.cmd

```diff
@@ -51,6 +56,25 @@ set "PYTHONNOUSERSITE=1"
 set "PATH=%DBT_TOOLCHAIN_ROOT%\python;%DBT_TOOLCHAIN_ROOT%\python\Scripts;%DBT_TOOLCHAIN_ROOT%\arm-none-eabi-gcc\bin;%DBT_TOOLCHAIN_ROOT%\openocd\bin;%DBT_TOOLCHAIN_ROOT%\cmake\bin;%DBT_TOOLCHAIN_ROOT%\ninja-build\bin;%PATH%"
 set "PROMPT=(dbt) %PROMPT%"
 
+if "%DBT_TOOLCHAIN_INSTALLED%" == "1" (
+    if [%DBT_NO_PYTHON_UPGRADE%] == [] (
+        echo DBT: Toolchain freshly installed, running python setup
+        set PIP_CMD=python -m pip
+        set "PIP_REQUIREMENTS_PATH=%SCRIPT_PATH%\scripts\toolchain\requirements.txt"
+        python -c "import wheel"
+        if %ERRORLEVEL% NEQ 0 (
+            set "PIP_WHEEL_PATH=%DBT_TOOLCHAIN_ROOT%\python\wheel"
+            for /R %PIP_WHEEL_PATH% %%G in (
+                certifi*.whl
+            ) do (
+                %PIP_CMD% install -q "%%G"
+            )
+        )
+        %PIP_CMD% install -q --upgrade pip | find /V "already satisfied"
```

Comment:
I'm seeing some errors, and I don't know enough to troubleshoot them. Sometimes, the script leaves a `win32-x86_64-10.zip` file and `win32-x86_64` directory in the project root, but I'm not sure how to repro that.

PowerShell:

```
PS C:\Users\Cowboy\projects\DelugeFirmware> .\dbt
Downloading Windows toolchain..done!
Extracting Windows toolchain..moving..done!
Cleaning up temporary files..done!
DBT: Toolchain freshly installed, running python setup
'install' is not recognized as an internal or external command,
operable program or batch file.
```

Git Bash:

```
$ ./dbt.cmd
Downloading Windows toolchain..done!
Extracting Windows toolchain..moving..done!
Cleaning up temporary files..done!
DBT: Toolchain freshly installed, running python setup
install: unknown option -- q
Try 'install --help' for more information.
find: '/V': No such file or directory
find: 'already satisfied': No such file or directory
install: unknown option -- q
Try 'install --help' for more information.
find: '/V': No such file or directory
find: 'already satisfied': No such file or directory
usage: dbt [-h] [<subcommand>]

positional arguments:
  <subcommand>

options:
  -h, --help    print this help message

subcommands:
  Building:
    build [-h] [-S] [-v] [-c] [-t TYPE] [-m] [{release,debug,relwithdebinfo,all}]
      Build the Deluge firmware
    clean [-h]
      Cleanup any build artifacts in the current environment
    configure [-h] [-f] [-m] [-t {dev,nightly,alpha,beta,rc,release}]
      Configure a CMake build (automatically called by build)
    nuke [-h]
      Indiscriminately removes any build tree contents and artifacts
    scons [-h]
      Run SCons in the current environment (deprecated)
  Debugging:
    debug [-h] [-j] [-c COMMAND] [-d TARGET_DEVICE] [-g GDB_PORT] [-v] [-p {swd,jtag}] [-i INTERFACE]
      Run a debug server (JLink or OpenOCD)
    gdb [-h] [-nb] [{7seg,oled}]
      Open a GDB remote for the proper target
    openocd [-h] [-v]
      Run OpenOCD with default arguments (CMSIS-DAP/DelugeProbe).
      These can be changed using the OPENOCD_OPTS variable.
  Development:
    docs [-h] [-R]
      Open the docs for the Deluge Firmware (rebuilding if necessary)
    shell [-h]
      Open a shell in the DBT environment
  Miscellaneous:
    format [-h] [-nr] [-q] [-v] [-c] [directory]
      Formats files using clang-format (Assumes .clang-format present in directory structure)
    license [-h] [-r] [-q] [-d] [-v] directory
      Add Synthstrom Deluge license prelude to files (if not present)
  Ungrouped:
    test [-h]
      Run deluge tests
```